packages/core/src/services/chains/run.ts

```diff
@@ -256,3 +297,16 @@ async function runStep({
     throw error
   }
 }
+
+function checkValidType(aiResult: AIReturn<StreamType>) {
+  const { type } = aiResult
+  const invalidType = type !== 'text' && type !== 'object'
+  if (!invalidType) return Result.nil()
+
+  return Result.error(
+    new ChainError({
+      code: RunErrorCodes.UnsupportedProviderResponseTypeError,
+      message: `Invalid stream type ${type} result is not a textStream or objectStream`,
+    }),
+  )
+}
```

Comment:
We can put it in another file if it helps. But this is actually better, because it splits responsibilities and makes the code easier to parse. 

This:
```
// chains/run.ts

checkValidType(...)
parseResponse(...)
createProviderLog(...)
```

Is better than this:
```
// chains/run.ts
providerProcessor(...)

// providerProcessor.ts
checkValidType(...)
parseResponse(...)
createProviderLog(...)
```